clippy_lints/src/methods/unnecessary_iter_cloned.rs

```diff
@@ -124,6 +124,9 @@ pub fn check_for_loop_iter(
                 };
                 diag.span_suggestion(expr.span, "use", snippet, applicability);
                 for addr_of_expr in addr_of_exprs {
+                    if addr_of_expr.span.from_expansion() {
```

Comment:
... I think even better would be to store a `Vec<(Span, String)>` in the visitor directly and then simply call `diag.multipart_suggestion()` here once with that vec as an argument.

A multipart suggestion would be better than what it is currently doing with repeated `diag.span_suggestion` anyway because it would avoid duplicating the same help message multiple times.